common/notification/types.ts

```diff
@@ -34,6 +34,8 @@ export interface NotificationContextExtensions {
     campaign?: string;
     // For Campaigns, support sending custom Mailjet Notifications:
     overrideMailjetTemplateID?: string;
+    // For Achievements, the match or subcourse is needed as a relation to allocate events to a specific user achievement
+    relation?: string;
```

Comment:
If we have no access to the relation, we would not be able to access it at all. Having a nullable relation field is the best solution we had. Do you have any other suggestion how we could make this work otherwise?

Reply:
If we did not include the relation to this NotificationContext the Error Property 'relation' does not exist on type 'SpecificNotificationContext<ID>' pops up.